docs/guide/CSS.md

```diff
@@ -468,3 +468,92 @@ For instance, if we have a widget with a (CSS) class called `dialog`, we could r
 
 Note that `initial` will set the value back to the value defined in any [default css](./widgets.md#default-css).
 If you use `initial` within default css, it will treat the rule as completely unstyled.
+
+
+## Nesting CSS
+
+!!! tip "Added in version 0.47.0"
+
+CSS rule sets may be *nested*, i.e. they can contain other rule sets.
+When a rule set occurs within an existing rule set, it inherits the selector from the enclosing rule set.
+
+Let's put this in to practical terms.
+The following example will display two boxes containing the text "Yes" and "No" respectively.
+These could eventually form the basis for buttons, but for this demonstration we are only interested in the CSS.
+
+=== "nesting01.tcss (no nesting)"
+
+    ```sass
+    --8<-- "docs/examples/guide/css/nesting01.tcss"
+    ```
+
+=== "nesting01.py"
+
+    ```python
+    --8<-- "docs/examples/guide/css/nesting01.py"
+    ```
+
+=== "Output"
+
+    ```{.textual path="docs/examples/guide/css/nesting01.py"}
+    ```
+
+The CSS is quite straight forward; there is a rule for the container, all buttons, and a rule each for the yes and no buttons.
+However it is easy to imagine this stylesheet growing more rules as we add features.
+
+Nesting allows us to group rule sets which have common selectors.
+In the example above, the rules all start with `#questions`, which is a good indication that we can nest the rules.
```

Comment:
This sets the expectation for nesting of the `#questions`, but not the `#questions .button`.
Would it be worth adding one sentence to say something like

```suggestion
In the example above, the rules all start with `#questions`, which is a good indication that we can nest the rules.
We can also see that the last three rules all start with `#questions .button`, which means we can nest the rules even further.
```

Reply:
Changed the wording to suggest it is a general principle, and not just for this one thing